src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -379,49 +412,62 @@ module Node = struct
     get_best_chain ?max_length ~logger t
     |> Deferred.bind ~f:Malleable_error.or_hard_error
 
-  let make_get_balance ~f ~logger t ~account_id =
+  type account_data =
+    { nonce : Mina_numbers.Account_nonce.t
+    ; total_balance : Currency.Balance.t
+    ; liquid_balance_opt : Currency.Balance.t option
+    ; locked_balance_opt : Currency.Balance.t option
+    }
+
+  let get_account_data ~logger t ~public_key =
     let open Deferred.Or_error.Let_syntax in
     [%log info] "Getting account balance"
       ~metadata:
-        ( ("account_id", Mina_base.Account_id.to_yojson account_id)
+        ( ("pub_key", Signature_lib.Public_key.Compressed.to_yojson public_key)
         :: logger_metadata t ) ;
-    let pk = Mina_base.Account_id.public_key account_id in
-    let token = Mina_base.Account_id.token_id account_id in
-    let get_balance_obj =
-      Graphql.Get_balance.make
-        ~public_key:(Graphql_lib.Encoders.public_key pk)
-        ~token:(Graphql_lib.Encoders.token token)
+    (* let pk = Mina_base.Account_id.public_key account_id in *)
+    (* let token = Mina_base.Account_id.token_id account_id in *)
+    let get_account_obj =
+      Graphql.Get_account_data.make
+        ~public_key:(Graphql_lib.Encoders.public_key public_key)
+        (* ~token:(Graphql_lib.Encoders.token token) *)
         ()
     in
-    let%bind balance_obj =
+    let%bind account_data_obj =
       exec_graphql_request ~logger ~node:t ~query_name:"get_balance_graphql"
-        get_balance_obj
+        get_account_obj
     in
-    match balance_obj#account with
+    match account_data_obj#account with
     | None ->
         Deferred.Or_error.errorf
-          !"Account with %{sexp:Mina_base.Account_id.t} not found"
-          account_id
+          !"Account with public_key %s not found"
```

Comment:
Any specific reason not show account_id? There could be accounts with same pk but different token_id and we may add them in the future

Reply:
but graphql only cares about the public key right?

Reply:
NM i brought the account id back into this function